### PR TITLE
[APIV4] POST /logs

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
This PR implements the POST /logs for apiV4

#### Ticket Link
n/a

#### Checklist
- [x] Added or updated unit tests
- [x] Added API documentation - PR: https://github.com/mattermost/mattermost-api-reference/pull/218
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api4/system.go [hunk 1]
```diff
@@ -160,6 +161,34 @@ func getLogs(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.ArrayToJson(lines)))
 }
 
+func postLog(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !*utils.Cfg.ServiceSettings.EnableDeveloper && !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
```
Comment (grundleborg) [inline api4/system.go]: Is it intentional that logging javascript errors into the server logs for non-sysadmin users only works in dev mode?

Comment (jwilander) [inline api4/system.go]: Yes, I asked for that specifically. I don't think we should be polluting customer logs with JS errors unless the have developer mode on, also it lets users spam the logs if they want which I think we should avoid.

Open to discussion on it though

Comment (cpanato) [inline api4/system.go]: I agree with @jwilander

Comment (grundleborg) [inline api4/system.go]: ok that's fine then. Just checking it was a deliberate decision.

**Comment by jwilander** [comment]
Looks good, just needs a rebase

**Comment by cpanato** [comment]
done @jwilander

**Comment by grundleborg** [comment]
Thanks @cpanato - that's merged now.

**Pull request merged**